src/main/java/edu/ucar/unidata/rosetta/controller/TemplateController.java

```diff
@@ -64,29 +50,302 @@
+     * redirected to step 2.  Otherwise, the user is taken pack to step 1 to correct the invalid data.
+     *
+     * @param data      The form-backing object.
+     * @param result    The BindingResult for error handling.
+     * @param model     The Model object to be populated.
+     * @param request   HttpServletRequest needed to pass to the dataManager to get client IP.
+     * @param response  HttpServletResponse needed for setting cookie.
+     * @return          Redirect to next step.
+     */
+    @RequestMapping(value = "/step1", method = RequestMethod.POST)
+    public ModelAndView processCFType(Data data, BindingResult result, Model model, HttpServletRequest request, HttpServletResponse response) {
+
+        if (result.hasErrors()) {   // validation errors
+            logger.error("Validation errors detected in CF type form data.");
+            model.addAttribute("error", result.getGlobalError().getDefaultMessage());
+            // Add current step to the Model (used by view to keep track of where we are in the wizard).
+            model.addAttribute("currentStep", "1");
+            // Add a list of all steps to the Model for rendering left nav menu.
+            model.addAttribute("steps", resourceManager.loadResources().get("steps"));
+            // Add domains data to Model (for platform display).
+            model.addAttribute("domains", resourceManager.loadResources().get("domains"));
+            // Add platforms data to Model (for platform selection).
+            model.addAttribute("platforms", resourceManager.loadResources().get("platforms"));
+            return new ModelAndView("wizard");
+        } else {
+            // Persist the data.
+            dataManager.persistData(data, request);
+            response.addCookie(new Cookie("rosetta", data.getId()));
+            return new ModelAndView(new RedirectView("/step2", true));
+        }
+    }
+
+
+    /**
+     * Accepts a GET request for access to step 2 of the wizard.
+     *
+     * @param model  The Model object to be populated.
+     * @param request   HttpServletRequest needed to get the cookie.
+     * @return  View and the Model for the wizard to process.
+     */
+    @RequestMapping(value = "/step2", method = RequestMethod.GET)
+    public ModelAndView displayFileUploadForm(Model model, HttpServletRequest request) {
+
+        // Have we visited this page before during this session?
+        Cookie rosettaCookie = WebUtils.getCookie(request, "rosetta");
+
+        // Create a Data form-backing object.
+        Data data;
+        if (rosettaCookie != null)
+            // Populate Data object with info.
+            data = dataManager.lookupById(rosettaCookie.getValue());
+        else
+            // Something has gone wrong.  We shouldn't be at this step without having persisted data.
+            throw new IllegalStateException("No persisted data available for file upload step.  Check the database & the cookie.");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "No persisted data available for file upload step.  Check the database & the cookie." 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)